### DISCUSSION TITLE
Inquiry about FHI-aims setting

### DISCUSSION THREAD

**Discussion by keysongkang** [mzjb/DeepH-pack#52] [Q&A]
Hello everyone,

Since I could reproduce the tutorial, I am revisiting my FHI-aims calculations.
After that, I come up with some questions.

1. When we print out 'basis-indices.out', which option should we use for 'control.in'?
    I used 'output matrices_2005'. But, I realize that when I use 'output aitranss' I get the same file with a different format.
    I want to make sure that I am using the suitable options.

2. When I compare the results in 'hamiltonian.out' and 'NoTB.dat', I found that the first component of Hamiltonian is different. Is it because of the unit conversion? Or, should it be the same? I got -65.623254 for 1,1 in 'hamiltonian.out' while I got 7143.47659470431 for first entry of hamiltonian in 'NoTB.dat'.

3. Does "spinful" option in 'info.json' appear only when we include the SOC effect?

Thank you.

**Comment by keysongkang** [comment]
Furthermore, I believe I have identified the issue with FHI-aims.

To assess the band structure results, I utilized the actual hamiltonian.h5 and overlap.h5 files. This implies that I employed the hamiltonian.h5 and overlap.h5 files obtained from the self-consistent field (SCF) calculation and executed the 'dense_calc.jl' script with these files. However, the results I obtained were incorrect.

This suggests that either the eigen_solver or the output in NoTB.dat is flawed. Nevertheless, since I was able to reproduce the tutorial results successfully, it appears that the eigen_solver is functioning properly.

Consequently, the problem seems to lie with the output in NoTB.dat. At this point, I am unsure of the next steps to take. Could you provide guidance on what should be done in this situation?
Would you be able to check that you have the same problem?

**Reply by keysongkang** [reply]
Thank you so much. I think I solve the problem. Now I can get the right band structure.

<img width="568" alt="image" src="https://github.com/mzjb/DeepH-pack/assets/130345610/0958c384-6c3e-4319-80d3-4792a951ebcf">

This would be my last question.
It seems that the current version of DeepH is only capable to calculate the band structure. Would it be possible to calculate the quantities such as electric susceptibility in the paper? If not, is there any plan to do that in the future?

**Comment by mzjb** [comment]
Hi,

1. I'm not familiar with FHI-aims, so I don't have a good understanding of the behavior of the 'basis-indices.out' file. And I haven't used the 'matrices_2005' parameter before, so I can't provide an answer to this question.
2. 'NoTB.dat' uses eV as its unit, and I'm not certain if 'hamiltonian.out' is the built-in Hamiltonian output file of FHI-aims. It's possible that 'hamiltonian.out' represents the Hamiltonian in k-space, while 'NoTB.dat' provides the real-space Hamiltonian required by DeepH, which could explain the difference. Another possibility is a unit conversion issue.
3. 'spinful' being set to 'True' indicates that non-collinear spin-polarized calculations were performed, and including SOC effects is one scenario where this can happen.
4. DeepH is a code designed solely for obtaining the Hamiltonian. In its design, it is used in conjunction with other post-processing code to derive properties from the Hamiltonian. The inclusion of band calculation functionality is mainly for the convenience of reproducing the results in the paper. In our paper, the results for calculating electric susceptibility and shift current, were obtained through a combined usage of DeepH and HopTB.jl (https://github.com/HopTB/HopTB.jl/issues/7#issuecomment-1528645008)."

I hope this helps clarify some aspects of your questions.
